resources/api-gateway/templates/monitoring.yaml

```diff
@@ -0,0 +1,51 @@
+{{- if and (.Capabilities.APIVersions.Has "monitoring.coreos.com/v1") (.Capabilities.APIVersions.Has "authentication.istio.io/v1alpha1") }}
+---
+# Dedicated Service for metrics endpoint
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "api-gateway.name" . }}-metrics
+  labels:
+{{ include "api-gateway.labels" . | indent 4 }}
+spec:
+  ports:
+    - name: metrics
+      port: {{ .Values.config.ports.metrics }}
+  selector:
+    app.kubernetes.io/name: {{ include "api-gateway.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    prometheus: monitoring
+{{ include "api-gateway.labels" . | indent 4 }}
+  name: {{ include "api-gateway.name" . }}
+spec:
+  endpoints:
+  - port: metrics
+    metricRelabelings:
+    - sourceLabels: [ __name__ ]
+      regex: ^(go_gc_duration_seconds|go_goroutines|go_memstats_alloc_bytes|go_memstats_heap_alloc_bytes|go_memstats_heap_inuse_bytes|go_memstats_heap_sys_bytes|go_memstats_stack_inuse_bytes|go_threads|http_requests_total|process_cpu_seconds_total|process_max_fds|process_open_fds|process_resident_memory_bytes|process_start_time_seconds|process_virtual_memory_bytes|rest_client_request_latency_seconds_bucket|rest_client_requests_total|workqueue_adds_total|workqueue_depth|workqueue_queue_duration_seconds_bucket)$
+      action: keep
+  namespaceSelector:
+    matchNames:
+      - "{{ .Release.Namespace }}"
+  selector:
+    matchLabels:
+{{ include "api-gateway.labels" . | indent 6 }}
+---
+apiVersion: authentication.istio.io/v1alpha1
+kind: Policy
```

Comment:
Please remove sidecar (add annotation), remove policy and create DR instead. We don't need a sidecar for this k8s controller